File: DSAlgorithms/Leetcode/sliding_window/assorted/c++/209-min_size_subarr_sum.cpp
class Solution {
public:
    int minSubArrayLen(int target, vector<int>& nums) {
       int left = 0;
       int total = 0;
       int result = std::numeric_limits<int>::max();
       for(int right = 0; right < nums.size(); right++)
       {
           total += nums[right];
           while(total >= target)
           {
               result = std::min(right - left + 1, result);
               total -= nums[left];
               left++;
           }
       }
       if(result == std::numeric_limits<int>::max())
            return 0;
        return result;
    }
};

File: dsa_algo/leetcode/ts/src/sliding_window/209-min_size_subarr_sum.ts
const minSubArrayLen = (target: number, nums: number[]): number =>{
    let left = 0;
    let total = 0;
    let result = Infinity;
    for(let right = 0; right < nums.length; right++)
    {
        total += nums[right];
        while(total >= target)
        {
            result = Math.min(right - left + 1, result);
            total -= nums[left];
            left++;
        }
    }
    if(result === Infinity)
        return 0;
    return result;
};